src/main/java/com/team766/orin/GetOrinRawValue.java

```diff
@@ -0,0 +1,100 @@
+ * Instead, it asks for a default value to return. This is a problem because we can't know if the value is not found or if the value is the default value.
+ * Thus, numbers were chosen as default values that aren't really plausible to be sent by the Orin.
+ * For example, it is not likley that the Orin will send Integer.MIN_VALUE as a value. In this case, an exception is thrown.
+ *
+ * @author Max Spier, 9/28/2024
+ */
+public class GetOrinRawValue {
+
+    private static NetworkTableInstance inst = NetworkTableInstance.getDefault();
+    private static NetworkTable table = inst.getTable("/SmartDashboard");
+
+    public static int getInt(String key) throws ValueNotFoundOnTableError {
+        IntegerTopic topic = table.getIntegerTopic(key);
+        IntegerEntry value = topic.getEntry(Integer.MIN_VALUE);
+        if (value.get() == Integer.MIN_VALUE) {
+            throw new ValueNotFoundOnTableError(key);
+        }
+        return (int) value.get();
+    }
+
+    public static double getDouble(String key) throws ValueNotFoundOnTableError {
+        DoubleTopic topic = table.getDoubleTopic(key);
+        DoubleEntry value = topic.getEntry(Double.NEGATIVE_INFINITY);
+        if (value.get() == Double.NEGATIVE_INFINITY) {
+            throw new ValueNotFoundOnTableError(key);
+        }
+        return value.get();
+    }
+
+    /*
+     * Be careful to know that the boolean value is valid. If the value is not found, the default value is returned and no exception is thrown.
+     * This is because true/false are the only two possible values for a boolean - if one were to be the default it would be hard to know if the value was not found or if the value was the default.
+     */
+    public static boolean getBoolean(String key) {
+        BooleanTopic topic = table.getBooleanTopic(key);
+        BooleanEntry value = topic.getEntry(false);
+        return value.get();
+    }
+
+    public static String getString(String key) throws ValueNotFoundOnTableError {
+        StringTopic topic = table.getStringTopic(key);
+        StringEntry value =
+                topic.getEntry(
+                        "Team 766 is such a great team and they will go to worlds every year because they are such a great team!!!");
+        if (value.get()
+                .equals(
+                        "Team 766 is such a great team and they will go to worlds every year because they are such a great team!!!")) {
+            throw new ValueNotFoundOnTableError(key);
+        }
+        return value.get();
+    }
+
+    public static int[] getIntArray(String key) throws ValueNotFoundOnTableError {
+        IntegerArrayTopic topic = table.getIntegerArrayTopic(key);
+        long[] arr = new long[1];
+        arr[0] = Integer.MIN_VALUE;
+        IntegerArrayEntry values = topic.getEntry(arr);
+        if (values.get()[0] == Integer.MIN_VALUE && values.get().length == 1) {
```

Comment:
check for the length first. that way if the array is empty, the `&&` will short-circuit and this won't try to access element 0 (ArrayIndexOutOfBoundsException)

Reply:
ah, yes, thanks!